### ISSUE TITLE
Python Linting and Outline stopped working.

### ISSUE THREAD

**Issue by BluesummersC** [microsoft/pylance-release#5463]
Type: <b>Bug</b>

I use VSCode with Remote-SSH extension to connect to a remote VM and use a custom python interpreter on that VM. I've been using this setup for at least 2 years with minimal issues.

Today, I connected to the VM as usual, opened my code, but do not get any linting or Symbols in the Outline. When I used it last week, it was working fine.

What I've tried: 
Re selected the interpreter.
Closed and reloaded VSCode.
Deleted ~/.vscode-server on the remote VM and re-connected and re-installed Python/Pylance to remote.
Uninstalled VSCode on local machine.
Rolled back to VSCode 1.86.0 on local machine.

So far, nothing has worked. 

Extension version: 2024.2.1
VS Code version: Code 1.86.0 (05047486b6df5eb8d44b2ecd70ea3bdf775fd937, 2024-01-31T10:28:19.990Z)
OS version: Windows_NT x64 10.0.22621
Modes:
Remote OS version: Linux x64 4.18.0-372.9.1.el8.x86_64

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-10750H CPU @ 2.60GHz (12 x 2592)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: enabled_on<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_graphite: disabled_off<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled|
|Load (avg)|undefined|
|Memory (System)|31.82GB (18.31GB free)|
|Process Argv|--crash-reporter-id 2107b15a-b4cb-4a83-89bb-4fab5441bff9|
|Screen Reader|no|
|VM|57%|

|Item|Value|
|---|---|
|Remote|SSH: vm-build|
|OS|Linux x64 4.18.0-372.9.1.el8.x86_64|
|CPUs|Intel(R) Xeon(R) CPU E5-2665 0 @ 2.40GHz (2 x 2400)|
|Memory (System)|5.61GB (3.12GB free)|
|VM|100%|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vspor879:30202332
vspor708:30202333
vspor363:30204092
vscod805cf:30301675
binariesv615:30325510
vsaa593cf:30376535
py29gd2263:30899288
c4g48928:30535728
azure-dev_surveyone:30548225
962ge761:30959799
pythongtdpath:30769146
welcomedialogc:30910334
pythonidxpt:30866567
pythonnoceb:30805159
asynctok:30898717
pythontestfixt:30902429
pythonregdiag2:30936856
pyreplss1:30897532
pythonmypyd1:30879173
pythoncet0:30885854
pythontbext0:30879054
dsvsc016:30899300
dsvsc017:30899301
dsvsc018:30899302
b1063948:30958189
e3gdj431:30958358

```

</details>

**Comment by debonte** [comment]
Can you please provide a log as described under "Filing an issue" in [the troubleshooting guide](https://github.com/microsoft/pylance-release/blob/main/TROUBLESHOOTING.md#filing-an-issue)?

**Comment by BluesummersC** [comment]
I thought I had 1.86.1 installed on the VM, but I had 1.68.1 instead. When I upgraded to 1.86.1 on the VM and updated Python and PyLance, the Outline instantly stopped working locally on the VM. It had been working fine locally on the VM since my last comment. So now both PC A and VM have VsCode 1.86.1 and neither could load the linting or symbols. 

I normally load into a folder that is a bunch of symlinks to various project directories, since my job requires jumping between frequently. The Python Language Server had an error saying that there was too much data or too many symlinks. I closed that folder and opened a single project directory and PyLance started working again. My takeaway is something changed in a recent update with how PyLance traverses symlinks or large directories, but I don't have the time or energy to figure out what, so I'll just only open one project directory at a time.
